src/com/google/enterprise/adaptor/ValidatedUri.java

```diff
@@ -70,12 +70,11 @@ public ValidatedUri(String uriString) throws URISyntaxException {
       throw new URISyntaxException(uriString, "relative URIs are not allowed");
     }
 
-    if (uri.getHost() == null) {
+    if (Strings.isNullOrEmpty(uri.getHost())) {
       throw new URISyntaxException(uriString, "no host");
     }
 
-    if ((Strings.isNullOrEmpty(uri.getRawPath())
-            || uri.getRawPath().equals("/"))
+    if (Strings.isNullOrEmpty(uri.getRawPath())
```

Comment:
I think this is trying to be too clever. One thing the individual connectors should perhaps do is make sure that there are parameters in the configured pattern. By the way, did you rip out the code checking for MessageFormat parameters?